addons/crm/models/crm_stage.py

```diff
@@ -48,3 +48,21 @@ def default_get(self, fields):
     @api.depends('team_id')
     def _compute_team_count(self):
         self.team_count = self.env['crm.team'].search_count([])
+
+    def write(self, vals):
+        """ Since leads that are in a won stage must have their
+        probability = 100%, this override ensures that setting a stage as won
+        will set all the lead in that stage to probability = 100%.
+        Inversely, if a won stage is not marked as won anymore, the lead
+        probability should be recomputed base on automated probability.
+        Note: If a user set a stage as won and change his mind right after,
+        the manuel probability will be lost in the process."""
```

Comment:
Few typos here:

base -> based
a user set -> a user sets
and change -> and changes
the manuel -> the manual